mypy/suggestions.py

```diff
@@ -653,7 +652,7 @@ def extract_from_decorator(self, node: Decorator) -> FuncDef | None:
             for ct in typ.items:
                 if not (
                     len(ct.arg_types) == 1
-                    and isinstance(ct.arg_types[0], TypeVarType)
+                    # and isinstance(ct.arg_types[0], TypeVarType)
                     and ct.arg_types[0] == ct.ret_type
                 ):
                     return None
```

Comment:
This is looks like a special case for this decorator:
```py
def f(x: T) -> T:
  return x
```

you should add a special case *in addition to* this.

Reply:
Well, technically this line is *limiting to* that special case. And this will even fix the linked issue, two callables of that shape will compare equal AFAIC. 

But this also makes the following (completely wrong) deco generate a signature:

```python
def deco(fn: str) -> str: ...

@deco
def foo():
    print()
```

So I agree there should be one more special case, not just blanket "one-arg function returning its arg unchanged".

Reply:
I would argue that any identity function should be fine here -- if that decorator happens to have the wrong types that's not really this code's problem (it's just trying to unwrap the underlying callable) and I suspect there's many many ways to type an identity function with varying specificity making this particular condition difficult-to-impossible to write correctly